Websocket Rust client keeps sending ping messages after connection is closed

# Bug Report

### Expected Behavior
The websocket automatically reconnects or the heartbeat task is canceled. The dYdX client has called the `disconnect()` method, but the heart beat task keeps running in the background.

### Actual Behavior
The websocket is closed by the venue:
```
nautilus_network::websocket: Received error message - terminating: WebSocket protocol error: Connection reset without closing handshake
```
The heartbeat task keeps sending ping messages which continuously fails. This is the first error message.
```
[1m2024-10-02T22:20:15.323098159Z[0m [ERROR] FIREHOSE-SPOT-001.nautilus_network::websocket: Error sending ping: Trying to work with closed connection
```
Somewhat later, the logs contain the same error message multiple times (at every ping interval)
```
1m2024-10-02T22:20:45.832380494Z[0m [ERROR] FIREHOSE-SPOT-001.nautilus_network::websocket: Error sending ping: WebSocket protocol error: Sending after closing is not allowed
```

The dYdX websocket client detects that ping messages are not being received anymore, and goes through a reconnection cycle called by `_reconnect_ping()`. That works successfully fortunately, but the heartbeat task of the previous websocket client is still running in the background.

### Specifications

- OS platform: AWS Fargate, Linux Docker
- Python version: 3.11
- `nautilus_trader` version: develop branch

--- a/nautilus_core/network/src/websocket.rs
+++ b/nautilus_core/network/src/websocket.rs
@@ -444,7 +444,12 @@ impl WebSocketClient {
                         }
                         break;
                     }
-                    (true, false) => break,
+                    // Close the heartbeat task on disconnect if the connection is already closed
+                    (true, false) => {
+                        tracing::debug!("Inner client is disconnected");
+                        tracing::debug!("Shutting down inner client to clean up running tasks");
+                        inner.shutdown().await
+                    }
                     _ => (),
                 }
             }

